sdk/peering/Azure.ResourceManager.Peering/tests/Scenario/PeeringAsnTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Peering.Models;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Peering.Tests
+{
+    internal class PeeringAsnTests : PeeringManagementTestBase
+    {
+        private SubscriptionResource _subscription;
+        private Random _random => new Random();
+        private PeerAsnCollection _peerAsnCollection => _subscription.GetPeerAsns();
+
+        public PeeringAsnTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public async Task TestSetUp()
+        {
+            _subscription = await Client.GetDefaultSubscriptionAsync();
+        }
+
+        [TearDown]
+        public async Task TestTearDown()
+        {
+            var list = await _peerAsnCollection.GetAllAsync().ToEnumerableAsync();
+            foreach (var item in list)
+            {
+                await item.DeleteAsync(WaitUntil.Completed);
+            }
+        }
+
+        [RecordedTest]
+        public async Task CreateOrUpdate()
+        {
+            string peerAsnName = Recording.GenerateAssetName("peerAsn");
+            var peerAsn = await CreatePeerAsn(peerAsnName);
+            ValidatePeeringService(peerAsn);
+            Assert.AreEqual(peerAsnName, peerAsn.Data.Name);
+            Assert.AreEqual(peerAsnName, peerAsn.Data.PeerName);
```

Comment:
I think we can add these two lines in the method `ValidatePeeringService`, just need to pass one more parameter `peerAsnName`.